app/js/controllers/main.js

```diff
@@ -1,10 +1,28 @@
-/* global Controller, Gesture */
+/*global EditView*/
+/*global ViewSourceView*/
+/*global ActionMenuView*/
+/*global SettingsView*/
+/*global AppendChildView*/
+/*global MoveView*/
+/*global MainView*/
+
+/*global EditController*/
+/*global ViewSourceController*/
+/*global ActionMenuController*/
+/*global SettingsController*/
+/*global AppendChildController*/
+/*global MoveController*/
+
+/*global Controller*/
+/*global Gesture*/
 
 export default class MainController extends Controller {
   constructor(options) {
     super(options);
 
     this._waitToBeOpened();
+
+    this._loadedModules = false;
```

Comment:
This pattern is very not-Promisey. I find this much clearer, and less prone to error:
```js
this._initialized = new Promise((resolve, reject) => {
  // eventually
  resolve();
});

// ...

this._initialized.then(() => {
  // do something that requires initialization
});

// ...

getSomeAsyncValueThatRequiresInit() {
  return new Promise((resolve, reject) => {
    this._initialized.then(() => {
      // do stuff
      resolve();
    });
  });
}
```